src/internal.h

```diff
@@ -111,4 +111,13 @@ int map_add_preimage_and_hash(struct wally_map *map_in,
 const struct wally_map_item *map_find_equal_integer(const struct wally_map *lhs,
                                                     const struct wally_map *rhs,
                                                     uint32_t key);
+
+/* Clamp input/output allocation sizing to standard tx sizes for BTC.
+ * Liquid numbers are smaller; we use the upper limit */
+#define TX_MAX_INPUTS_ALLOC 1738u
```

Comment:
@instagibbs see merchs answer here: https://bitcoin.stackexchange.com/questions/35570/what-is-the-maximum-number-of-inputs-outputs-a-transaction-can-have

Note that these aren't hard maximums, you can create txs with more than these numbers of inputs and outputs. This is just the maximum we will optimize preallocation for.

Reply:
Sure, was just curious because my numbers were not matching up. Inputs can actually be smaller than that, but they wouldn't be "payment" inputs, so under the reasoning, totally ok.